tests/hw5/oop1_test.py

```diff
@@ -0,0 +1,67 @@
+import datetime
+
+import pytest
+
+from homework5 import oop_1
+
+
+@pytest.fixture()
+def sample_for_test():
+    """
+    Pull of sample data for test
+    :return: 2 objects of class Homework, object of class Student and Teacher
+    """
+    student = oop_1.Student("Ivanov", "Ivan")
+    teacher = oop_1.Teacher("Smirnov", "Lev")
+    homework_1 = oop_1.Homework("Maths", 5)
+    homework_2 = oop_1.Homework("Statistics", 0)
+    return homework_1, homework_2, student, teacher
+
+
+def test_class_homework(sample_for_test):
+    """
+    Testing that attributes and method of Class Homework work correct
+    """
+    homework_1, homework_2, _, _ = sample_for_test
+    text_1, deadline_1, is_active_1 = (
+        homework_1.text,
+        homework_1.deadline,
+        homework_1.is_active(),
+    )
+    text_2, deadline_2, is_active_2 = (
+        homework_2.text,
+        homework_2.deadline,
+        homework_2.is_active(),
+    )
+    assert text_1 == "Maths"
+    assert deadline_1 == datetime.timedelta(days=5)
+    assert text_2 == "Statistics"
+    assert deadline_2 == datetime.timedelta(0)
+    assert is_active_1 is True
+    assert is_active_2 is False
```

Comment:
Получилось очень много кода только для того чтобы потом в `assert` писать поменьше. В случае когда у нас используется прямой доступ к полю объекта не обязательно создавать переменную. Это может быть оправдано если имена объектов и полей слишком длинные, в нашем случае всё довольно лаконично, так что вполне можно ассертить напрямую:
```suggestion
    assert homework_1.text == "Maths"
    assert homework_1.deadline == datetime.timedelta(days=5)
    assert homework_2.text == "Statistics"
    assert homework_2.deadline == datetime.timedelta(0)
    assert homework_1.is_active() is True
    assert homework_2.is_active() is False
```

аналогичная ситуация в следующем тесте с `last_name`, а вот в последнем хоть и `first_name` выглядит также излишней, но `create_hw_1` и `create_hw_2` вполне уместны т.к. и код длиннее и взятие переменной происходит уже из 3го подуровня.